include/net/download_client.h

```diff
@@ -266,6 +287,34 @@ int download_client_file_size_get(struct download_client *client, size_t *size);
  */
 int download_client_disconnect(struct download_client *client);
 
+/**
+ * @brief Download a file asynchronously.
+ *
+ * This initiates a asynchronous connect-download-disconnect sequence to the target
+ * host. When only one file is required form a target server, it can be used instead of
+ * separate calls to download_client_set_host(), download_client_start()
+ * and download_client_disconnect().
+ *
+ * The download is carried out in fragments of up to
+ * @kconfig{CONFIG_DOWNLOAD_CLIENT_HTTP_FRAG_SIZE} bytes for HTTP, or
+ * @kconfig{CONFIG_DOWNLOAD_CLIENT_COAP_BLOCK_SIZE} bytes for CoAP,
+ * which are delivered to the application
+ * via @ref DOWNLOAD_CLIENT_EVT_FRAGMENT events.
+ *
+ * @param[in] client	Client instance.
+ * @param[in] host	Name of the host to connect to, null-terminated.
+ *			Can include scheme and port number, defaults to
+ *			HTTP or HTTPS if no scheme is provided.
+ * @param[in] config	Configuration options.
+ * @param[in] file	File to download, null-terminated.
```

Comment:
No need for a separate host and file. You can use the complete URI for both the server and the filename.

Reply:
Well, many of the existing FOTA APIs separate file and host, so why change it here?

Reply:
The download client can already separate the host name from the file name for you. So no need to suggest that the user also needs to do this.
If they already have one (or more) file name(s) and the host name separate they can still use the `_set_host()` (multiple) `_start()` `_disconnect()` scheme.
A function like wget also does not require you to split the uri yourself.

Reply:
Fixed. This accepts both forms now.